src/devices/video/upd7220.cpp

```diff
@@ -999,58 +985,155 @@ void upd7220_device::draw_rectangle(int x, int y)
 //  draw_char -
 //-------------------------------------------------
 
-void upd7220_device::draw_char(int x, int y)
+void upd7220_device::draw_char()
 {
-	int isize,psize;
-	uint16_t tile_data;
+	const int8_t dir_change[2][4] = {
+		{2, 2, -2, -2},
+		{1, 3, -3, -1}
+	};
+	const uint8_t type = (m_figs.m_figure_type & 0x10) >> 4;
 
-	LOG("uPD7220 char check: %d %d %02x %08x %d %d %02x\n",x,y,m_figs.m_dir,m_ead,m_figs.m_d,m_figs.m_dc,m_figs.m_figure_type);
+	uint16_t pattern;
 
-	/* QX10 may require upper bits for psize, VT240 requires the opposite */
-	isize = m_figs.m_d;
-	psize = m_figs.m_dc + 1;
+	LOG("uPD7220 char check: %08x %04x %02x %02x %d %d %02x\n",m_ead,m_mask,m_bitmap_mod,m_figs.m_dir,m_figs.m_d,m_figs.m_dc,m_figs.m_figure_type);
 
-	for(int pi = 0; pi < psize; pi++)
+	for (int i = 0, di = 0; i < (m_figs.m_dc + 1); ++i)
```

Comment:
The rest of the file has no space between flow control keywords and the opening parentheses of the control expressions.  Please stick with that within the file.

Reply:
This isn't the case though, a quick look through the file shows it is already quite inconsistent with regards to a space/nospace between keywords and parentheses. I understand the desire for consistency, but in this case the file already uses a mix of both.